### ISSUE TITLE
$$phase of null

### ISSUE THREAD

**Issue by felixSchl** [Reactive-Extensions/rx.angular.js#52]
Every now and then I would get a `cannot access $$phase of null` exception using the scope scheduler.

Using the following seems to get around this, although I am not sure if this is the right way to do it:

```js
(scheduler._scope.$$phase || (scheduler._scope.$root && scheduler._scope.$root.$$phase))
        ? fn()
        : scheduler._scope.$apply(fn);
```

Refer to [scopescheduler.js](https://github.com/Reactive-Extensions/rx.angular.js/blob/master/src/scopescheduler.js#L37)

Also, I was wondering if it might be due to a $scope that was destroyed. Is there a way to have the scheduler dispose of the subscription when the $scope gets destroyed?

**Comment by felixSchl** [comment]
This is resolved by #53

**Comment by PhiLhoSoft** [comment]
For the record, I got this error as well, because we forgot a dispose of a subscribe in one controller. So the subscription was still there, but the controller supposed to handle it was destroyed, so was its scope.
